src/Compiler/FSComp.txt

```diff
@@ -1697,4 +1698,5 @@ featureEscapeBracesInFormattableString,"Escapes curly braces before calling Form
 3565,parsExpectingType,"Expecting type"
 featureInformationalObjInferenceDiagnostic,"Diagnostic 3559 (warn when obj inferred) at informational level, off by default"
 3566,tcMultipleRecdTypeChoice,"Multiple type matches were found:\n%s\nThe type '%s' was used. Due to the overlapping field names\n%s\nconsider using type annotations or change the order of open statements."
-3567,parsMissingMemberBody,"Expecting member body"
\ No newline at end of file
+3567,parsMissingMemberBody,"Expecting member body"
+3568,tcConstraintIntersectionSyntaxUsedWithNonFlexibleType,"Constraint intersection syntax may only be used with flexible types."
```

Comment:
I feel like the term "flexible types" might be unknown for some.
I am think if the message should either have a specific example, or point to https://learn.microsoft.com/en-us/dotnet/fsharp/language-reference/flexible-types